### PR TITLE
Add Wayland icon

### LINKED ISSUE #5056: Wayland (display server protocol)
**Brand Name:** Wayland
**Website:** https://wayland.freedesktop.org/
**Alexa rank:** ~3.1 miliion (as of 2021-02-18)
  
It is attempting to replace Xorg (which is also on the icon set), some of them are being developed by the Xorg developers.
It is slowly being developed and implemented in different desktops (e.g., Mutter in GNOME desktop, KDE Plasma was recently making massive progress especially in its [recent release](https://kde.org/announcements/plasma/5/5.21.0/)).
Also, its most popular implementation of the protocol, [sway](https://github.com/swaywm/sway) has 8.5k stars (as of 2021-02-18) so I think it paints a good idea of the popularity.

**Official resources for icon and color:**
- SVG file is at https://gitlab.freedesktop.org/wayland/weston/-/blob/master/data/wayland.svg
- Color is `#ffbc00` only because of the logo.

### PR DESCRIPTION
**Pull request by valpackett**
wayland

**Issue:** closes #5056

**Similarweb rank:** :shrug: 

### Checklist
  - [x] I updated the JSON data in `_data/simple-icons.json`
  - [x] I optimized the icon with SVGO or SVGOMG
  - [x] The SVG `viewbox` is `0 0 24 24`

### Description

See #5056

### CODE REVIEW

**Review by sisp — COMMENTED** [review]
Thanks for your contribution, @unrelentingtech! :pray:

I wonder whether there is a way to retain the shapes outside the circle. Currently, everything outside the circle is removed. Since I can't find an official monochromatic icon for Wayland, we might need to come up with something ourselves.

I imagine we could add inset strokes for the missing part of the "W" above the circle. And perhaps the sprinkles above the circle could be filled shapes. WDYT, @unrelentingtech  and also @simple-icons/icon-review?

**Comment by unrelentingtech** [comment]
Sprinkles would look okay filled (inverted) but I think the parts of the W would look awkward no matter what, so I went with just the stuff inside the circle.

**Comment by sisp** [comment]
Okay, I agree. Would you mind updating icon to include the inverted sprinkles?

**Comment by valpackett** [comment]
test:

waylandtest

I don't think this is better. The extra sprinkles are silly, they don't add much but make the circle necessarily smaller.

**Comment by sisp** [comment]
Agreed. That actually doesn't look good.

I'll accept your proposed icon, just found some differences between the original one and yours:



Would you checking your conversion steps?

**Review by sisp — CHANGES_REQUESTED** [review]
Agreed. That actually doesn't look good.

I'll accept your proposed icon, just found some differences between the original one and yours:



Would you mind checking your conversion steps?

**Comment by valpackett** [comment]
I used Inkscape to subtract the `W` path from the circle path and resize the result to fit the 24x24 viewbox, and used SVGO to trim the path precision down to 3 decimals. I think. I might have actually involved rsvg-convert to resize the viewbox specifically which would've been a bit silly in retrospect. These don't seem like major differences :shrug: but I can redo everything I guess

**Comment by LitoMore** [comment]
@valpackett Could you try with precision 4? You could refer to this https://github.com/simple-icons/simple-icons/pull/7908#issuecomment-1266943734.

**Comment by valpackett** [comment]
Updated with higher precision

File: icons/wayland.svg [hunk 1]
```diff
@@ -0,0 +1 @@
...7832.4786v.0898c-.4076-.0608-.6939-.1747-.8594-.3418-.1783.4102-.4141 1.5494-.707 3.418-.2293.866-.5156 1.2988-.8594 1.2988-.968-.3798-1.7763-1.0012-2.4258-1.8672-.1019 0-.2688-.4329-.498-1.2988-.1401 0-.2343-.4329-.2852-1.2988-.484-1.2306-.8159-2.6369-.9942-4.2168v-2.3692h-.0761l-.3438.0899v-.0899c0-.1063.1145-.1601.3438-.1601l-.0762-1.3672V7.664l-.5547-3.418v-.621zm7.125.8027v.0684c-.2674.243-.7003.4627-1.2988.6601v-.1581c.1019 0 .1543-.0848.1543-.252l1.1445-.3184zm10.8067.1309-.0313.2559h.1328l.0567-.045a12 12 0 0 0-.1582-.211zm-7.1758.666.0762.0918v.0684l-.1348.1816h-.0762l-.1523-.1816v-.0684l.287-.0918zm2.6347.0918h.0782v.0684l-.1348.1816h-.0762v-.0898l.1328-.1602zm-1.203 1.1387.287.2285c-.051.1671-.1907.25-.4199.25l-.1523-.25c0-.076.094-.1525.2851-.2285zm1.7206 1.0488.0567.0918v.0684l-.0567.0898h-.1543v-.0898l.1543-.1602zm-2.7129.8203v.0684l-.3633.3418h-.2109v-.0918c0-.1367.1922-.2424.5742-.3184zm-11.8437 3.168h.0762v.0684c-.191.5013-.358.8038-.498.9101h-.1524v-.3418l.5742-.6367zm9.3398 3.8047h.0782v.1601l-.5743.3418-.0761-.0918v-.1582l.5722-.252zM4.291 16.369h.1328l.2285.8887H4.5l-.209-.8203v-.0684zm7.6406 1.047h.0567l.2304.2519-.078.0683h-.0762l-.1329-.2285v-.0918z"/></svg>
```
Comment (sisp) [inline icons/wayland.svg:1]:
```suggestion
<svg role="img" viewBox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><title>Wayland</title><path d="M23.982 12c0 9.224-9.985 14.989-17.973 10.377A11.982 11.982 0 0 1 .018 12c0-3.386 1.346-6.306 3.444-8.418v.675l.553 3.413v1.07l.077 1.365c-.23 0-.344.053-.344.159v.09l.344-.09h.076v2.366c.178 1.578.509 2.981.992 4.21.05.864.146 1.297.286 1.297.229.864.394 1.297.496 1.297a5.494 5.494 0 0 0 2.422 1.865c.344 0 .63-.432.859-1.296.292-1.866.528-3.004.706-3.414.165.167.451.281.858.342v-.091c0-.152-.26-.311-.782-.478.14-.728.33-1.35.572-1.866.038-.88.49-2.1 1.354-3.663l.344.09c.05.714.146 1.305.286 1.776.521 1.061.782 2.063.782 3.003 0 .364.407 1.585 1.22 3.663.204 1.366.465 2.17.783 2.412h.152c.42 0 .916-.538 1.488-1.615.56-1.214 1.011-1.889 1.354-2.025 0-.106-.12-.16-.362-.16v-.09c.19 0 .286-.054.286-.16v-.068l-.076-.091.801-.319c0-.273-.095-.41-.286-.41v-.09h.572v-.16c-.292 0-.483-.159-.572-.477h.133v-.319l-.076-.091c.432 0 .649-.083.649-.25v-.069h-.134v-.09c.153 0 .267-.16.343-.478-.42-.137-.629-.273-.629-.41l.553-1.615c0-.82.146-1.715.439-2.685.33-.94.566-1.965.706-3.072.304-1.064.617-1.787.938-2.17A11.979 11.979 0 0 1 23.982 12Zm-5.99-10.377c.587.34 1.138.724 1.65 1.148a33.45 33.45 0 0 0-.157 1.736c-.458 1.077-.96 2.594-1.507 4.55h-1.411l-.725.07v.181h.362l.286-.09 1.202.158v.092c-.343 1.183-.553 1.774-.63 1.774-.445 2.215-.8 3.322-1.068 3.322h-.229c-.318-.288-.674-.88-1.068-1.774a.861.861 0 0 0-.21-.57L13.42 9.855c-.216-1-.502-1.752-.858-2.252-.115-.334-.305-.5-.572-.5h-.134c-.42 0-.826.295-1.22.887l.076.25c-.178.607-.465 1.312-.859 2.116-.61 1.775-1.01 2.662-1.201 2.662h-.153l-.286-.227c-.483-1.305-.814-3.224-.992-5.757h-.134v-.16h.058a1.33 1.33 0 0 1 .515-.75v-.069l-.077-.09-.076.09-.076-.09-.344.09h-.152c-.09 0-.23-.348-.42-1.046v-.182h.496v-.16h-.648c-.139-.698-.545-1.53-1.218-2.494.296-.206.601-.398.915-.577h.016l.077.068.438-.068h.783c.228 0 .464.083.705.25.115-.152.33-.235.649-.25v-.16h-2.38c3.11-1.667 6.947-1.99 10.51-.392l-.443.142v.16c.24 0 .4-.095.482-.284.368.166.734.354 1.095.562Zm3.477 3.203h-.134l.035-.295.179.23-.08.065ZM9.719 1.595h.572l.133.159.363-.16.42.069.438-.068v-.16H9.72v.16Zm3.7 1.297v.159l.21.09v-.09l.076-.091c.14 0 .21-.053.21-.16v-.068h-.21l-.286.16Zm-1.144.887h-.058l-.076-.068-.42.227v.16c.61-.137.916-.266.916-.387V3.62l-.362.159Zm-2.842.978c0 .167-.051.25-.153.25v.16c.598-.197 1.03-.417 1.297-.66V4.44l-1.144.318Zm4.482.57v.068l.153.182h.076l.134-.182v-.069l-.077-.09-.286.09Zm2.785.159v.09h.076l.134-.181v-.069h-.076l-.134.16Zm-1.354 1.206.152.25c.23 0 .37-.084.42-.25l-.286-.228c-.19.076-.286.152-.286.228Zm1.85.978v.091h.153l.057-.091v-.068l-.057-.091-.153.159Zm-3.128.978v.091h.21l.362-.34v-.07c-.382.077-.572.183-.572.32ZM2.24 12.13v.341h.153c.14-.106.305-.41.496-.91v-.068h-.076l-.573.637Zm9.328 3.413v.159l.076.091.572-.341v-.16h-.076l-.572.25Zm-7.267.887.21.82h.152l-.229-.888h-.133v.068Zm7.63 1.07.133.227h.076l.076-.068-.228-.25h-.058v.09Z"/></svg>
```

**Review by sisp — CHANGES_REQUESTED** [review]
Thanks for the updated icon. Unfortunately, there are still notable differences when I compare it with the original one:


When I derive the icon, my steps are:
1. Ungroup all paths.
2. Subtract the "W" (+ sprinkles) path from the circle path.
3. Change the height to 24 while retaining the aspect ratio. The aspect ratio of the original icon is not exactly 1:1, its height is slightly larger than its width.
4. Change the viewport to 24x24 and center the path.

I perform all conversion steps with the highest possible precision in Boxy SVG.

When I compare this icon with optimized versions using SVGOMG, I get the following difference with a precision of 3

and with a precision of 4:


There are some tiny negligible differences with a precision of 3 and practically no differences with a precision of 4.

I've suggested the path data of my derivation with a precision of 3 in case you're interested.

**Comment by valpackett** [comment]
oh, sorry. Okay, merged your suggestion, now the linter complains that it's `23.964 x 23.999` and no dimension is exactly 24. oops..

**Review by sisp — CHANGES_REQUESTED** [review]
@valpackett Apologies for my negligence. I've made tiny change in the path data and `svglint` now passes locally for me.

**Comment by valpackett** [comment]
Thanks, merged everything now

**Pull request merged**